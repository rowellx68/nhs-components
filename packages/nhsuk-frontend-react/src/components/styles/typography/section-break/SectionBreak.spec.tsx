import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { composeStory } from '@storybook/react';
import meta, {
  Default as DefaultStory,
  Visible as VisibleStory,
} from './SectionBreak.stories';

const Default = composeStory(DefaultStory, meta);
const Visible = composeStory(VisibleStory, meta);

it('should render the SectionBreak component', () => {
  const { container } = render(<Default />);

  expect(container.querySelector('.nhsuk-section-break')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it('should render the SectionBreak component with visible', () => {
  const { container } = render(<Visible />);

  expect(
    container.querySelector('.nhsuk-section-break--visible'),
  ).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
