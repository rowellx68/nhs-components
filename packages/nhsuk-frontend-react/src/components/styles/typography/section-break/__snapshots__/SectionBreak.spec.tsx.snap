// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the SectionBreak component 1`] = `
<div>
  <hr
    class="nhsuk-section-break"
  />
</div>
`;

exports[`should render the SectionBreak component with visible 1`] = `
<div>
  <hr
    class="nhsuk-section-break nhsuk-section-break--visible"
  />
</div>
`;
